src/AspNetCore.Authorization/HttpAuthorizationHelper.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authorization.Policy;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.Server.AspNetCore.Authorization
+{
+    public static class HttpAuthorizationHelper
+    {
+        /// <summary>
+        /// Authorize current user based on provided options
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="policyName"></param>
+        /// <returns>True if the user was authorized and the request should proceed; False if the request should be ended immediately.</returns>
+        public static async Task<bool> AuthorizeAsync(HttpContext context, string policyName)
```

Comment:
I would suggest modelling this after the `[Authorize]` attribute in ASP.NET Core. In that attribute you can pass:

- AuthenticationSchemes - (This is a comma delimeted string but only because you can't pass an array to an attribute). I think this one is important to provide an overload for.
- RoleName - I would recommend using policies but some people like to shoot themselves in the foot and use role names.
- ActiveAuthenticationSchemes - This is obsolete. Do not add this.

Also, is the `authorization` repo dead after this? I literally submitted a PR yesterday to use the ASP.NET Core Authorization package! Basically, what this PR is all about.

Reply:
I think it's best to keep this simple. You can create a policy that specifies the authentication schemes or require role(s). I used a role check for the tests in Transports.AspNetCore.Tests\AuthenticatedTestStartup.cs.

I think the stuff in the authorization repo is more for authorizing individual nodes. I felt there was a separate need to support transport authorization that integrates with ASP.NET Core authentication and will trigger the authentication challenge or forbidden responses.